prog.go

```diff
@@ -185,6 +209,25 @@ func newProgramWithOptions(spec *ProgramSpec, opts ProgramOptions, handles *hand
 		if err != nil {
 			return nil, fmt.Errorf("load target BTF: %w", err)
 		}
+	} else if spec.AttachTarget != nil {
```

Comment:
I think this `else if` branch needs to move into `spec.AttachTo != ""`.
* opts.TargetBTF specifies the location of the vmlinux BTF (for environments where that isn't available from a convenient location)
* Even if opts.TargetBTF is specified, we need to use spec.AttachTarget to resolve the type ID for freplace programs.

```
targetBTF = nil
if opts.TargetBTF {
    targetBTF = ...
}

if spec.BTF {
    btf.ProgramFixups(spec.BTF, targetBTF)
}

if spec.AttachTo {
    if spec.AttachTarget {
        targetBTF = ...
    }
    resolveBTFType(targetBTF)
}
```

1. btf.ProgramFixups is always invoked with the kernel BTF or nil
2. resolveBTFType is invoked with the kernel BTF, the AttachTarget BTF or nil

The implication is that freplace programs are CO-RE relocated against the kernel, not against the program they are attaching to. Is that the correct behaviour?